subsys/net/ip/tcp_private.h

```diff
@@ -263,6 +264,7 @@ struct tcp { /* TCP connection */
 	enum tcp_data_mode data_mode;
 	uint32_t seq;
 	uint32_t ack;
+	uint16_t recv_win_max;
```

Comment:
Since you added the recv_win_max, is it possible to restructure the code so that the recv_win is not required any more and based on the ack value and buffer filling? This removes another state variable with the potential of going out of sync.

Reply:
Could you elaborate what do you mean by "buffer filling"? I don't really see what you suggest.

Reply:
We can't use `queue_recv_data` as it's used for a different purpose - the TCP stack queues out-of-order-packets in this queue, it has nothing to do with the window size.

There's a separate fifo where packets are queued before being passed to the application (`recv_data`), but I don't really see how could it be used to track the current window size. Once the packet is passed to the application, it's lost from the TCP stack radar, however the window shouldn't be updated until the packet is consumed by the application (i. e. at the socket/net_context layer). So I'm afraid we still need a variable to track the current window size.